[xla hash update] update the pinned xla hash

ezyang opened the pull request:
This PR is auto-generated nightly by [this action](https://github.com/pytorch/pytorch/blob/master/.github/workflows/_update-commit-hash.yml).
Update the pinned xla hash.

ezyang commented:
LGTM. CI failure seems spurious.

zdevito commented on torch/csrc/jit/ir.h:
```diff
@@ -571,6 +574,9 @@ Node * NodeWithKind<Self,K>::allocClone(Graph * in_graph) {
     } break; \
     case NodeKind::Kind: { \
       Kind * value = static_cast<Kind*>(__match_key); (void) value;
+#define IR_ELSEIF_NOCAST(Kind) \
```
Is this just a performance optimization?

apaszke replied:
No, it's because we have `torch::jit::Add` and `torch::autograd::Add`, so the compiler complains because of `static_cast<Add*>`.

zdevito commented on torch/csrc/jit/test_jit.cpp:
```diff
@@ -77,6 +77,14 @@ static void codeTemplateTest() {
 
 #ifdef WITH_CUDA
+template<typename T>
+Node * appendNewNode(Graph& graph, ArrayRef<Node*> inputs) {
```
Don't primitives already have an init that does this?

zdevito replied:
Or is this related to the fact that std::initializer_list and perfect forward don't work together?

apaszke replied:
It's related to the fact that primitives have trivial constructors (don't accept a list of inputs). It's just to make the change simpler, and tests smaller (we'd need this loop after every node otherwise)

ezyang committed:
Generalize update-xla-commit-hash and setup torchdynamo hash

The hash isn't used for anything yet.

Signed-off-by: Edward Z. Yang <ezyangfb.com>

ghstack-source-id: ecd1a4f8e57c93d759b365ab0165609954272e27
Pull Request resolved: https://github.com/pytorch/pytorch/pull/79042

ezyang committed:
boop

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

pytorchmergebot committed:
update xla commit hash
